bgpd/bgp_ecommunity.c

```diff
@@ -625,9 +627,10 @@ static const char *ecommunity_gettoken(const char *str,
 					goto error;
 			} else {
 				/* ASN */
-				as = strtoul(buf, &endptr, 10);
-				if (*endptr != '\0' || as == BGP_AS4_MAX)
+				tmp_as = strtoull(buf, &endptr, 10);
+				if (*endptr != '\0' || tmp_as > BGP_AS4_MAX)
```

Comment:
One question though: should ecommunity work with asdot/asdot+ formats? :)

Reply:
I'm not sure if/how asdot works - I haven't examined any of those codepaths.  The existing `soo` CLI doesn't look like it interprets `x.y:z` as `4-byte:2-byte` encoding, even with the `as-notation dot` config in place:
```
ub20(config)# router bgp 1 as-notation dot
ub20(config-router)# neighbor 1.1.1.1 remote-as external 
ub20(config-router)# add ipv4 uni
ub20(config-router-af)# neighbor 1.1.1.1 soo 16.16:1
ub20(config-router-af)# do sh run | include soo
  neighbor 1.1.1.1 soo 16.0.0.16:1      <<<<<  parser assumes "16.16" is an ipv4 address since it includes "."
```

This PR isn't attempting to bridge that gap, just to fix a bug in the existing functionality.
I think adding parser functionality for dot/dot+ notation should probably come via a separate PR where the time/effort has been spent to cover all the cases for the as-dot feature.

Reply:
No, it was just a question (for discussion) :smile:

Reply:
yeah... I'm not really sure how asdot notation is supposed to work wrt to ecommunity parsing/encoding.
I'd guess this same function would need some work, since this is where we do string -> eval conversion... :shrug: